### PR TITLE
Updates for ingress

### PR DESCRIPTION
**Pull request by binq**
Updates for ingress

### CODE REVIEW

**Review by jsimonVA — APPROVED** [review]

File: modules/vye/app/models/vye/user_profile.rb [hunk 1]
```diff
@@ -45,29 +47,40 @@
...
-  def self.produce(attributes)
-    attributes = attributes.slice(:ssn, :file_number, :icn)
+    user_profile = find_by(ssn_digest:) || find_by(file_number_digest:)
 
-    %i[ssn file_number].each do |key|
-      record = send("find_from_digested_#{key}", attributes[key])
+    if user_profile.blank?
+      user_profile = build(ssn_digest:, file_number_digest:, icn:)
+    else
+      user_profile.ssn_digest = ssn_digest if user_profile.ssn_digest.blank?
```
[the remark this answers is not in the document]
Comment (binq) [reply modules/vye/app/models/vye/user_profile.rb:59]: This is code not storing the information as clear-text. The attribute `ssn_digest` is a `one-way hash` of the clear-text the function was called with.

File: modules/vye/app/models/vye/user_profile.rb [hunk 2]
```diff
@@ -45,29 +47,40 @@
...
+    conflict = false
+    attribute_name = nil
 
-      return record
+    if new_record?
+      conflict = false
+    elsif ssn_digest != attributes['ssn_digest']
+      conflict = true
+      attribute_name = 'ssn'
+      self.ssn_digest = ssn_digest
```
[the remark this answers is not in the document]
Comment (binq) [reply modules/vye/app/models/vye/user_profile.rb:77]: This is code not storing the information as clear-text. The attribute `ssn_digest` is a `one-way hash` of the clear-text the function was called with.

**Review by stevenjcumming — CHANGES_REQUESTED** [review]
Please write a some unit tests for following methods of `Vye::UserProfile`

- `self.find_or_build`
- `check_for_match`

Also consider writing some tests for `LoadData#load_profile` to test the various conflict conditions

File: modules/vye/app/models/vye/user_profile.rb [hunk 3]
```diff
@@ -45,29 +47,40 @@ def self.find_and_update_icn(user:)
...
+    ssn_digest, file_number_digest = [ssn, file_number].map { |value| gen_digest(value) }
 
-  def self.produce(attributes)
-    attributes = attributes.slice(:ssn, :file_number, :icn)
+    user_profile = find_by(ssn_digest:) || find_by(file_number_digest:)
 
-    %i[ssn file_number].each do |key|
-      record = send("find_from_digested_#{key}", attributes[key])
+    if user_profile.blank?
+      user_profile = build(ssn_digest:, file_number_digest:, icn:)
```
[the remark this answers is not in the document]
Comment (binq) [reply modules/vye/app/models/vye/user_profile.rb]: > Where is `build` coming from?

`build` is coming from activerecord.

**Comment by binq** [comment]
> Please write a some unit tests for following methods of `Vye::UserProfile`
> 
> * `self.find_or_build`
> * `check_for_match`
> 
> Also consider writing some tests for `LoadData#load_profile` to test the various conflict conditions

These changes have been pushed.

LoadData#load_profile is being tested by the TIMS and BDN ingress scripts.

**Comment by jsimonVA** [comment]
@stevenjcumming Can you review this. Changes have been made. Thank you

**Comment by jsimonVA** [comment]
@jsimonVA  test comment

**Review by stevenjcumming — APPROVED** [review]

**Pull request merged**